### DISCUSSION TITLE
freedreno kgsl

### DISCUSSION THREAD

**Discussion by rhjdvsgsgks** [termux/termux-packages#16524] [General]
hi

i found that someone are working on freedreno kgsl backend (https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/21570) . just like turnip kgsl. it will talk to kernel by /dev/kgsl-3d0 and provide gles on it. without using vendor's driver.

on the mainline linux. freedreno can provide up to opengl 4.5 (depends on which device). if kgsl backend keep growing. maybe we can also have that for downstream kernel (android)

**Comment by rhjdvsgsgks** [comment]
seems like kgsl patches only avaliable on android platform. but mesa in termux need x11 platform. maybe we need to port https://gitlab.freedesktop.org/Hazematman/mesa/-/commit/7c84dff0ded2144db4356f7d3cc4404e93b13a93 to x11 platform

**Comment by nerdydumbykier** [comment]
I am also somewhat interested on how this would work.

Turnip and Zink together works well but requires special env like "MESA_VK_WSI_DEBUG=sw".

**Comment by JesusChapman** [comment]
![Screenshot_20250409-024817 Termux](https://github.com/user-attachments/assets/ff7c8665-c9f2-429d-8691-9fc0fa064fb5)

I've managed to get it working, but unfortunately it's not possible to run a desktop environment on it, it just corrupts the interface.
